pkg/auth/dependency/interaction/flows/anonymous.go

```diff
@@ -101,3 +101,18 @@ func (f *AnonymousFlow) Authenticate(requestJWT string, clientID string) (*authn
 
 	return attrs, nil
 }
+
+func (f *AnonymousFlow) DecodeUserID(requestJWT string) (string, anonymous.RequestAction, error) {
+	identity, request, err := f.Anonymous.ParseRequest(requestJWT)
```

Comment:
Seems `Anonymous.ParseRequest` may return nil identity if kid not exists and key is provided in jwt? Maybe we should check and return `ErrInvalidCredentials` if identity is nil here.